sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredential.java

```diff
@@ -75,13 +75,23 @@ public class AzurePipelinesCredential implements TokenCredential {
                     HttpRequest request = new HttpRequest(HttpMethod.POST, url);
                     request.setHeader(HttpHeaderName.AUTHORIZATION, "Bearer " + systemAccessToken);
                     request.setHeader(HttpHeaderName.CONTENT_TYPE, "application/json");
+                    request.setHeader(HttpHeaderName.fromString("X-TFS-FedAuthRedirect"), "Suppress");
```

Comment:
nit: For all `HttpHeaderName` usages make them constants. Using `HttpHeaderName.fromString` will result in a cache map lookup each time which is a bit of a waste of time.